deeppavlov/core/trainers/fit_trainer.py

```diff
@@ -0,0 +1,264 @@
+                # noinspection PyPackageRequirements
+                # noinspection PyUnresolvedReferences
+                import tensorflow
+            except ImportError:
+                log.warning('TensorFlow could not be imported, so tensorboard log directory'
+                            f'`{self.tensorboard_log_dir}` will be ignored')
+                self.tensorboard_log_dir = None
+            else:
+                self.tensorboard_log_dir = expand_path(tensorboard_log_dir)
+                self._tf = tensorflow
+
+        self._built = False
+        self._saved = False
+        self._loaded = False
+
+    def fit_chainer(self, iterator: Union[DataFittingIterator, DataLearningIterator]) -> None:
+        """
+        Build the pipeline :class:`~deeppavlov.core.common.chainer.Chainer` and successively fit
+        :class:`Estimator <deeppavlov.core.models.estimator.Estimator>` components using a provided data iterator
+        """
+        if self._built:
+            raise RuntimeError('Cannot fit already built chainer')
+        for component_index, component_config in enumerate(self.chainer_config['pipe'], 1):
+            component = from_params(component_config, mode='train')
+            if 'fit_on' in component_config:
+                component: Estimator
+
+                targets = component_config['fit_on']
+                if isinstance(targets, str):
+                    targets = [targets]
+
+                if self.batch_size > 0 and callable(getattr(component, 'partial_fit', None)):
+                    writer = None
+
+                    for i, (x, y) in enumerate(iterator.gen_batches(self.batch_size, shuffle=False)):
+                        preprocessed = self._chainer.compute(x, y, targets=targets)
+                        # noinspection PyUnresolvedReferences
+                        result = component.partial_fit(*preprocessed)
+
+                        if result is not None and self.tensorboard_log_dir is not None:
+                            if writer is None:
+                                writer = self._tf.summary.FileWriter(str(self.tensorboard_log_dir /
+                                                                         f'partial_fit_{component_index}_log'))
+                            for name, score in result.items():
+                                summary = self._tf.Summary()
+                                summary.value.add(tag='partial_fit/' + name, simple_value=score)
+                                writer.add_summary(summary, i)
+                            writer.flush()
+                else:
+                    preprocessed = self._chainer.compute(*iterator.get_instances(), targets=targets)
+                    if len(targets) == 1:
+                        preprocessed = [preprocessed]
+                    result: Optional[Dict[str, Iterable[float]]] = component.fit(*preprocessed)
+
+                    if result is not None and self.tensorboard_log_dir is not None:
```

Comment:
Have you considered making a separate method for wrinting to tensorboar?

Reply:
I did just now, but it looks like this method would have to have a lot of parameters and it would not make the code simpler.